12.0 website sale mandate

njeudy opened the pull request:
- Initial module that ask sepa info in extra step on ecommerce process

Commit:
[NEW] website_sale_mandate for SEPA direct debit

Commit:
[NEW] add controler and form for SEPA mandate in extra shop step

stephansainleger commented on website_sale_mandate/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+{
+    "name": "Ask SEPA mandate on checkout",
+    "version": "14.0.1.0.1",
```
we are in 12.0 :-)

Commit:
[TRANSLATE] add fr translation

stephansainleger commented on website_sale_mandate/views/sale_order_view.xml:
```diff
@@ -0,0 +1,24 @@
+                </page>
+            </xpath>
+            <button name="preview_sale_order" position="before">
+                <button name="action_create_partner_mandate" type="object" string="Create SEPA Mandate" class="btn-primary" attrs="{'invisible': [('show_mandate_action', '=', False)]}"/>
```
Improvement idea: have a "Pending validation" information on the sale order, in this case : 
- user has clicked on button "Create SEPA Mandate"
- the SEPA mandate is not validated yet

Because currently when we click on the button, there is no change on the screen, we do not know if the process has worked, and the button is still displayed (then it means we can launch the process several times in parallel).

The proposition here is that, if Odoo find a mandate for this partner with the same iban but not validated, it hides the "Create SEPA mandate" button and display a "Pending mandate validation" tag somewhere.

stephansainleger reviewed (changes requested):
One change required (version number), and an improvement idea.

Commit:
[FIX] fix version to 12.0

Commit:
[FIX] reload view and auto validate SEPA mandat

stephansainleger reviewed (approved):
Technical and functional review OK

clementthomas committed:
[IMP] odoo-usability-addons: change in extra-info
Change design and text of extra info step
By-pass extra info step if SEPA mandate already exists

clementthomas committed:
[IMP] odoo-usability-addons: bugfix on checkout + translation

stephansainleger commented:
New commits proposed by @clementthomas are validated.
